chess-AI/train.py

```diff
@@ -15,73 +23,144 @@ class MctsTrain:
         mcts: References the MCTS class
         policy_converter: References the PlayNetworkPolicyConverter class
     """
-    def __init__(self, mcts_simulations, exploration):
+    def __init__(self, mcts_simulations, exploration, lr):
         # Set the number of Monte Carlo Simulations
         self.mcts_simulations = mcts_simulations
 
-        self.training_examples = []
-        self.mcts = Mcts(exploration)
-        self.policy_converter = PlayNetworkPolicyConverter()
+        # Stores mcts probabilities and outcome reward values
+        self.mcts_probs = []
+        self.mcts_evals = []
 
-    def training_episode(self, nnet):
-        # At each episode need to set the training example to empty
-        self.training_examples = []
+        # Stores board fen strings
+        self.boards = []
 
-        # The board needs to be initialized ot the starting state
-        board = chess.Board()
-        fen_string = board.fen()
+        self.mcts = Mcts(exploration)
+        self.policy_converter = PlayNetworkPolicyConverter()
 
-        while True:
-            # Perform mcts simulation
-            for _ in range(self.mcts_simulations):
-                self.mcts.search(board, nnet)
+        # Learning rate
+        self.lr = lr
 
-            # Need to get the moves and policy from the mcts
-            # NOTE: moves[i] corresponds to search_probs[i]
-            moves, search_probs = self.mcts.find_search_probs(fen_string)
+    def training_episode(self, nnet, epochs):
+        with torch.no_grad():
+            # At each episode need to set the training example and network predictions to empty
+            self.mcts_probs = []
+            self.mcts_evals = []
+            self.boards = []
```

Comment:
Could add a comment describing that this is a 2d array where each 1d array consists of fen strings corresponding to one game